pkg/generate/git/repository.go

```diff
@@ -378,8 +379,7 @@ func timedCommand(timeout time.Duration, name, dir string, env []string, args ..
 
 	// check whether git was available in the first place
 	if err != nil {
-		_, err := exec.LookPath(name)
-		if err != nil {
+		if !isBinaryInstalled(name) {
 			return "", "", ErrGitNotAvailable
```

Comment:
i still find it weird that this timedCommand method is, in theory, generic, but then returns "ErrGitNotAvailable" if the command isn't found.  How would you feel about changing timedCommand to not make the command name configurable?  Or changing ErrGitNotAvailable to a more generic ErrCommandNotFound or something?